File: python/lesson01/solution/hello.py
import sys
import time
from lib.tracing import init_tracer


def say_hello(hello_to):
    with tracer.start_span('say-hello') as span:
        span.set_tag('hello-to', hello_to)

        hello_str = 'Hello, %s!' % hello_to
        span.log_kv({'event': 'string-format', 'value': hello_str})

        print(hello_str)
        span.log_kv({'event': 'println'})

assert len(sys.argv) == 2

tracer = init_tracer('hello-world')

hello_to = sys.argv[1]
say_hello(hello_to)

# yield to IOLoop to flush the spans
time.sleep(2)
tracer.close()

File: nodejs/lesson01/solution/hello.js
const assert = require("assert");
const { initTracer } = require("../../lib/tracing");

const tracer = initTracer("hello-world");

const sayHello = helloTo => {
  const span = tracer.startSpan("say-hello");
  span.setTag("hello-to", helloTo);
  const helloStr = `Hello, ${helloTo}!`;
  span.log({
    event: "string-format",
    value: helloStr,
  });
  console.log(helloStr);
  span.log({ event: "print-string" });
  span.finish();
};

assert.ok(process.argv.length == 3, "Expecting one argument");
const helloTo = process.argv[2];

sayHello(helloTo);

//tracer.close(callback gets called after all work is complete.  Likely need to use process.exit node call)

// tracer.close(function() {
//   process.exit();
// });
//Note: the above close call just closes right away, like it is sync, not async.


const closeTracer = () => tracer.close();

setTimeout(closeTracer, 12000);
